signer/src/bitcoin/utxo.rs

```diff
@@ -109,6 +109,64 @@ pub trait GetFees {
     fn get_fees(&self) -> Result<Option<Fees>, Error>;
 }
 
+/// Filter out the deposit requests that do not meet the amount or fee requirements.
+pub struct SbtcRequestsFilter<'a> {
+    sbtc_limits: &'a SbtcLimits,
+    minimum_fee: u64,
+}
+
+impl<'a> SbtcRequestsFilter<'a> {
+    /// Create a new [`DepositFilter`] instance.
+    pub fn new(sbtc_limits: &'a SbtcLimits, minimum_fee: u64) -> Self {
+        Self { sbtc_limits, minimum_fee }
+    }
+
+    /// Validate deposit requests based on four constraints:
+    /// 1. The user's max fee must be >= our minimum required fee for deposits
+    ///     (based on fixed deposit tx size)
+    /// 2. The deposit amount must be greater than or equal to the per-deposit minimum
+    /// 3. The deposit amount must be less than or equal to the per-deposit cap
+    /// 4. The total amount being minted must stay under the maximum allowed mintable amount
+    fn validate_deposit(
+        &self,
+        amount_to_mint: &mut Amount,
+        req: &'a DepositRequest,
+    ) -> Option<RequestRef<'a>> {
```

Comment:
This only validates the deposit amounts, so I would name this slightly different, so that it's clearer what is being validated.

At first I thought that this should return a boolean, but that would make it's use more annoying down the line so returning a `Option<RequestRef<'a>>` makes more sense.